### PR TITLE
feat(pubsub): Add Open Telemetry trace support

### PR DESCRIPTION
**Pull request by quartzmo**
Reference impl: googleapis/google-cloud-go#5034

closes: #13960

### CODE REVIEW

File: google-cloud-pubsub/google-cloud-pubsub.gemspec [hunk 1]
```diff
@@ -21,6 +21,7 @@ Gem::Specification.new do |gem|
   gem.add_dependency "concurrent-ruby", "~> 1.1"
   gem.add_dependency "google-cloud-core", "~> 1.5"
   gem.add_dependency "google-cloud-pubsub-v1", "~> 0.0"
+  gem.add_dependency "opentelemetry-sdk", "~> 1.0"
```
Comment (dazuma) [inline google-cloud-pubsub/google-cloud-pubsub.gemspec:24]: We cannot depend on opentelemetry-sdk. Use opentelemetry-api. At most the tests (development dependency) can depend on sdk.

Comment (quartzmo) [reply google-cloud-pubsub/google-cloud-pubsub.gemspec:24]: Understood. This was a mistake, only the tests depend on sdk. Thanks for catching.

Comment (quartzmo) [reply google-cloud-pubsub/google-cloud-pubsub.gemspec]: Fixed, thanks!

**Comment by quartzmo** [comment]
@hongalex PTAL, thanks!

**Comment by hongalex** [comment]
Closing due to staleness and the design has since been updated.

**Pull request closed without merging**